pkg/cache/application_events.go

```diff
@@ -47,6 +47,35 @@ func (st SimpleApplicationEvent) GetApplicationID() string {
 	return st.applicationID
 }
 
+// ------------------------
+// ApplicationStatusChangeEvent updates the status in the application CRD
+// ------------------------
+type ApplicationStatusChangeEvent struct {
+	applicationID string
+	event         events.ApplicationEventType
+	State         string
```

Comment:
for a state change, it needs to have 2 fields: `source` and `destination` state.
if we do not have this in the scheduler-interface, I think we need to add that too. Sorry for missing this earlier.

Reply:
From the application point of view we don't need the source state, because when this event is generated the transition is already done, so we just need to update it in the CRD. The destination field will be unused. I suggest to extend it with the destination only if we will need it.